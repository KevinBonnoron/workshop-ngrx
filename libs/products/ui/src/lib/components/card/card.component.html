<mat-card class="h-24" (click)="onClick()">
  <mat-card-header>
    <mat-card-title>{{ product.name }}</mat-card-title>
  </mat-card-header>
</mat-card>
