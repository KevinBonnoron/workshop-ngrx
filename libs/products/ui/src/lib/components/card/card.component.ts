import { Component, Input, inject } from '@angular/core';
import { MatCardModule } from '@angular/material/card';
import { Router } from '@angular/router';
import { Product } from '@workshop/shared';

@Component({
  standalone: true,
  imports: [MatCardModule],
  selector: 'products-ui-card',
  templateUrl: './card.component.html'
})
export class CardComponent {
  private readonly router = inject(Router);

  @Input()
  product!: Product;

  onClick() {
    this.router.navigate(['/edit/', this.product.id]);
  }
}
