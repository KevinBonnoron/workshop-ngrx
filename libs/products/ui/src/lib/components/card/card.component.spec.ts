import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Router } from '@angular/router';
import { RouterTestingModule } from '@angular/router/testing';
import { CardComponent } from './card.component';

class BlankComponent {}

describe('CardComponent', () => {
  let fixture: ComponentFixture<CardComponent>;
  let component: CardComponent;
  let router: Router;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule.withRoutes([{ path: 'edit/0', component: BlankComponent }]),
        CardComponent
      ]
    });

    fixture = TestBed.createComponent(CardComponent);
    component = fixture.componentInstance;
    router = TestBed.inject(Router);
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should call endpoint when onSave is called', () => {
    const routerNavigateSpy = jest.spyOn(router, 'navigate').mockImplementation(() => Promise.resolve(true));

    component.product = { id: 0, name: 'Test' };
    component.onClick();

    expect(routerNavigateSpy).toHaveBeenCalledWith(['/edit/', 0]);
  });

  it('should match snaptshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
