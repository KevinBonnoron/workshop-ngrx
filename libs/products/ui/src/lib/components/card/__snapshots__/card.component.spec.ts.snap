// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should match snaptshot 1`] = `
<products-ui-card
  router={[Function _Router]}
>
  <mat-card
    class="mat-mdc-card mdc-card h-24"
  >
    <mat-card-header
      class="mat-mdc-card-header"
    >
      <div
        class="mat-mdc-card-header-text"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
        </mat-card-title>
      </div>
    </mat-card-header>
  </mat-card>
</products-ui-card>
`;
